www/casestudies/items/customquestions/custom.php

```diff
@@ -0,0 +1,151 @@
+
+    $security = [
+    'user_id'      => $studentid,
+    'domain'       => $_SERVER['SERVER_NAME'],
+    'consumer_key' => $consumer_key,
+    'timestamp'    => gmdate('Ymd-Hi')
+    ];
+
+    $init = new Init('questions', $security, $consumer_secret, [
+        'id'                   => 'custom-shorttext',
+        'name'                 => 'Custom Short Text',
+        'course_id'            => $courseid,
+        'type'                 => 'local_practice',
+        'state'                => 'initial',
+        'session_id'           => $session_id
+        ]);
+
+    $request = '{
+    "response_id": "custom-shorttext-response-' . $session_id .'",
+    "type": "custom",
+    "js": "//' . $_SERVER['HTTP_HOST'] . '/casestudies/items/customquestions/custom_shorttext.js",
+    "css": "//' . $_SERVER['HTTP_HOST'] . '/casestudies/items/customquestions/custom_shorttext.css",
+    "stimulus": "What is the capital of Australia?",
+    "valid_response": "Canberra",
+    "score": 1
+}';
+
+?>
+
+<style>    
+    .custom-score {
+        position: absolute;
+        font-size: 17px;
+        margin-top: 5px;
+    }
+    .editor{
+        margin-bottom: 15px;
+    }
+</style>
+
+<div class="jumbotron section">
+     <div class="toolbar">
+        <ul class="list-inline">
+            <li data-toggle="tooltip" data-original-title="Visit the documentation"><a href="http://docs.learnosity.com/assessment/questions/knowledgebase/customquestions" title="Documentation"><span class="glyphicon glyphicon-book"></span></a></li>
+            <li data-toggle="tooltip" data-original-title="Toggle product overview box"><a href="#"><span class="glyphicon glyphicon-chevron-up jumbotron-toggle"></span></a></li>
+        </ul>
+    </div>
+    <div class="overview">
+        <h1>Custom Question - Short Text</h1>
+        <p>Here is a demo which shows an example custom implementation of the Short Text question type. You can rewrite the question JSON to define your own custom questions.</p>
+    </div>
+</div>
+
+<div class="section">        
+    <div class="row">
+        <div class="col-md-6">
+            <h2 class="page-heading">Question JSON</h2>
+            <div id="editor" class="editor"><?php echo htmlspecialchars($request);?></div>
+        </div>
+        <div class="col-md-6">
+            <h2 class="page-heading">Preview</h2>
+            <div id="custom_question_wrapper"></div>
+        </div>
+    </div>
+    <div class="row">
+        <div class="col-md-6">
+            <button class="btn btn-primary pull-right" id="render_custom_question">Render JSON</button>
+        </div>
+        <div class="col-md-6">
+            <div class="custom-score"><strong>Score: </strong> <span id="question_score">0</span> / <span id="question_max_score">0</span></div>
+            <button class="btn btn-primary pull-right" id="validate_question">Check Answer</button>
+        </div>
+    </div>
+</div>    
+
+<script src="//questions.learnosity.com"></script>
```

Comment:
We use variables for API URLs now, see any of the other demos. Means they work across all environments (vg, staging and prod).